<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>snRNA-seq data analysis</title>
  <meta name="description" content="This is a minimal benchmark for analyzing snRNA-seq data." />
  <meta name="generator" content="bookdown 0.30 and GitBook 2.6.7" />

  <meta property="og:title" content="snRNA-seq data analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal benchmark for analyzing snRNA-seq data." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="snRNA-seq data analysis" />
  
  <meta name="twitter:description" content="This is a minimal benchmark for analyzing snRNA-seq data." />
  

<meta name="author" content="Hong Jiang" />


<meta name="date" content="2022-11-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="preprocess.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#comments-on-snrna-seq-vs-scrna-seq"><i class="fa fa-check"></i><b>1.1</b> Comments on snRNA-seq vs scRNA-seq</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="preprocess.html"><a href="preprocess.html"><i class="fa fa-check"></i><b>2</b> Preprocess</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocess.html"><a href="preprocess.html#empty-droplets"><i class="fa fa-check"></i><b>2.1</b> Empty droplets</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="preprocess.html"><a href="preprocess.html#statistical-modeling"><i class="fa fa-check"></i><b>2.1.1</b> Statistical Modeling</a></li>
<li class="chapter" data-level="2.1.2" data-path="preprocess.html"><a href="preprocess.html#deep-learning"><i class="fa fa-check"></i><b>2.1.2</b> Deep Learning</a></li>
<li class="chapter" data-level="2.1.3" data-path="preprocess.html"><a href="preprocess.html#could-cellbender-retain-more-good-nucleus"><i class="fa fa-check"></i><b>2.1.3</b> Could Cellbender retain more good nucleus?</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="preprocess.html"><a href="preprocess.html#ambient-contamination"><i class="fa fa-check"></i><b>2.2</b> Ambient contamination</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">snRNA-seq data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">snRNA-seq data analysis</h1>
<p class="author"><em>Hong Jiang</em></p>
<p class="date"><em>2022-11-12</em></p>
</div>
<div id="preface" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Preface<a href="index.html#preface" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This is a minimal <strong>benchmark</strong> for analyzing snRNA-seq data, especially with scverse and bioconductor communities. It will be continuously updated throughout my PhD years.</p>
<p>From my current experience:</p>
<ul>
<li>bioconductor has more solid basement of biological (annotation) data</li>
<li>questions and methods to tackle them (except for dl) are usually first developed in R, then the python community would propose competing methods. Since seruat comes earlier than scanpy</li>
<li>the scanpy community evolves fast but somehow the developers are not considering better integrating each otherâ€™s new features</li>
<li>packages in R is in general easier to use, though I personally is more used to python and could adapt more myself</li>
<li>R utils are too easily to crash with the same hardware facility, same data size and same kind of job</li>
</ul>
<div id="comments-on-snrna-seq-vs-scrna-seq" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Comments on snRNA-seq vs scRNA-seq<a href="index.html#comments-on-snrna-seq-vs-scrna-seq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>advantages of snRNA-seq</strong>: does not require the preservation of cellular integrity during sample preparation, especially dissociation.</p>
<p><strong>disadvantages of snRNA-seq</strong>: loss of biological signal for genes with little nuclear localization.</p>
<p><strong>Interpretation of DE results</strong>:</p>
<ol style="list-style-type: decimal">
<li>assumption: nuclear abundances are a good proxy for the overall expression profile.</li>
<li>considerations:
<ul>
<li>transcripts for strongly expressed genes might localize to the cytoplasm for efficient translation and subsequently be lost upon stripping.</li>
<li>genes with the same overall expression but differences in <strong>the rate of nuclear export</strong> may appear to be differentially expressed between clusters.</li>
<li>In the most pathological case, higher snRNA-seq abundances may indicate <strong>nuclear sequestration</strong> of transcripts for protein-coding genes and reduced activity of the relevant biological process, <strong>contrary to</strong> the usual interpretation of the effect of upregulation.</li>
</ul></li>
</ol>
<!-- ## Usage  -->
<!-- Each **bookdown** chapter is an .Rmd file, and each .Rmd file can contain one (and only one) chapter. A chapter *must* start with a first-level heading: `# A good chapter`, and can contain one (and only one) first-level heading. -->
<!-- Use second-level and higher headings within chapters like: `## A short section` or `### An even shorter section`. -->
<!-- The `index.Rmd` file is required, and is also your first book chapter. It will be the homepage when you render the book. -->
<!-- ## Render book -->
<!-- You can render the HTML version of this example book without changing anything: -->
<!-- 1. Find the **Build** pane in the RStudio IDE, and -->
<!-- 1. Click on **Build Book**, then select your output format, or select "All formats" if you'd like to use multiple formats from the same book source files. -->
<!-- Or build the book from the R console: -->
<!-- ```{r, eval=FALSE} -->
<!-- bookdown::render_book() -->
<!-- ``` -->
<!-- To render this example to PDF as a `bookdown::pdf_book`, you'll need to install XeLaTeX. You are recommended to install TinyTeX (which includes XeLaTeX): <https://yihui.org/tinytex/>. -->
<!-- ## Preview book -->
<!-- As you work, you may start a local server to live preview this HTML book. This preview will update as you edit the book when you save individual .Rmd files. You can start the server in a work session by using the RStudio add-in "Preview book", or from the R console: -->
<!-- ```{r eval=FALSE} -->
<!-- bookdown::serve_book() -->
<!-- ``` -->

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="preprocess.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
