<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Preprocess | snRNA-seq data analysis</title>
  <meta name="description" content="This is a minimal benchmark for analyzing snRNA-seq data." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Preprocess | snRNA-seq data analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal benchmark for analyzing snRNA-seq data." />
  <meta name="github-repo" content="https://github.com/brainfo/thesisBook" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Preprocess | snRNA-seq data analysis" />
  
  <meta name="twitter:description" content="This is a minimal benchmark for analyzing snRNA-seq data." />
  

<meta name="author" content="Hong Jiang" />


<meta name="date" content="2023-04-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="rna-dynamics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Thesis Book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#comments-on-snrna-seq-vs-scrna-seq"><i class="fa fa-check"></i><b>1.1</b> Comments on snRNA-seq vs scRNA-seq</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="preprocess.html"><a href="preprocess.html"><i class="fa fa-check"></i><b>2</b> Preprocess</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocess.html"><a href="preprocess.html#empty-droplets"><i class="fa fa-check"></i><b>2.1</b> Empty droplets</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="preprocess.html"><a href="preprocess.html#statistical-modeling"><i class="fa fa-check"></i><b>2.1.1</b> Statistical Modeling</a></li>
<li class="chapter" data-level="2.1.2" data-path="preprocess.html"><a href="preprocess.html#deep-learning"><i class="fa fa-check"></i><b>2.1.2</b> Deep Learning</a></li>
<li class="chapter" data-level="2.1.3" data-path="preprocess.html"><a href="preprocess.html#could-cellbender-retain-more-good-nucleus"><i class="fa fa-check"></i><b>2.1.3</b> Could Cellbender retain more good nucleus?</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="preprocess.html"><a href="preprocess.html#ambient-contamination"><i class="fa fa-check"></i><b>2.2</b> Ambient contamination</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-dynamics.html"><a href="rna-dynamics.html"><i class="fa fa-check"></i><b>3</b> RNA dynamics</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">snRNA-seq data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="preprocess" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Preprocess<a href="preprocess.html#preprocess" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this chapter, we deal with empty droplets, ambient RNAs and doublets, especially in snRNA-seq setting.</p>
<div id="htmlwidget-af526dfa69ee0e9c9d20" style="width:672px;height:100px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-af526dfa69ee0e9c9d20">{"x":{"diagram":"\n  digraph graph2 {\n  \n  graph [layout = dot, rankdir = LR]\n  \n  # node definitions with substituted label text\n  node [shape = oval]\n  a [label = \"Cellbender\"]\n  b [label = \"emptyDrops\"]\n  c [label = \"controlAmbience\"]\n  d [label = \"enriched in empty drops\"]\n  \n  a -> c\n  a -> d\n  b -> c\n  b -> d\n  }","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<div id="empty-droplets" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Empty droplets<a href="preprocess.html#empty-droplets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have no prior knowledge about whether a particular library (i.e., cell barcode) corresponds to cell-containing or empty droplets. Thus, we need to call cells from empty droplets based on the observed expression profiles. This is not entirely straightforward as empty droplets can contain ambient (i.e., extracellular) RNA that can be captured and sequenced, resulting in non-zero counts for libraries that do not contain any cell.
The aim of empty droplets detection in most cases is to retains nucleus that would have been discarded by other threshold on total counts. Therefore, could use looser thresholds after this step.</p>
<div id="statistical-modeling" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Statistical Modeling<a href="preprocess.html#statistical-modeling" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Cell Ranger v3.0 and <a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html">DropletUtils</a> implemented utils to <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y">distinguish empty droplets</a>. The algorithm has three key steps:</p>
<ol style="list-style-type: decimal">
<li>filter out genes with counts of zero for all barcodes</li>
<li>estimate the profile of the ambient RNA pool (as gene count across barcodes with total counts &lt; 100 in DropletUtils or estimate expected number of recovered cells by OrdMag alg. in Cellranger)</li>
<li>test each barcode for deviations from this profile using a Dirichlet-multinomial model of UMI count sampling
<ul>
<li>Barcodes with significant deviations are considered to be genuine cells</li>
<li>Use a knee point filter to ensure that barcodes with large total counts are always retained</li>
</ul></li>
</ol>
<p>Cellranger uses an <code>expected-cells</code> and DropUtils uses <code>by.rank</code> as initial “cells”. But with the sample initial top number of cells, two implementations give slightly different retained cells. <em>And I don’t get why this happens.</em></p>
<div class="infobox notes">
<p>In cellranger 7.0, it estimates expected number of recovered cells by OrdMag alg. <span style="color: white;">But I didn’t get to use this method.</span><br />
Cellranger doesn’t explicitly tell what FDR cutoff it uses. But retained number of nucleus fell into the range between 0.001 to 0.05 of DropUtils.<br />
DropUtils uses barcodes with total counts &lt; 100 as initial empty drops, but this <strong>failed</strong> with the tested datasets, evident (e.g.) as in FDR distribution:</p>
</div>
</div>
<div id="deep-learning" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Deep Learning<a href="preprocess.html#deep-learning" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://cellbender.readthedocs.io/en/latest/">Cellbender</a> provides a deep generative model of background-contaminated counts. The model is used for learning the background RNA profile, distinguishing cell-containing droplets from empty ones, and retrieving background-free gene expression profiles.<br />
The model schema and network representation can be found in <a href="https://www.biorxiv.org/content/10.1101/791699v2.full.pdf">Figure 1g,h</a><br />
This method has been benchmarked with UMI counts threshold and DecontX but not DropletUtils</p>
</div>
<div id="could-cellbender-retain-more-good-nucleus" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Could Cellbender retain more good nucleus?<a href="preprocess.html#could-cellbender-retain-more-good-nucleus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>YES</strong><br />
I summarized the general stats from the above mentioned methods with three in-house 10x snRNA-seq data:</p>
<div class="infobox notes">
<p>Thresholds filtering is necessary and the most useful data clean method.<br />
It doesn’t matter whether by detecting outliers or just setting hard cut-offs.</p>
</div>
</div>
</div>
<div id="ambient-contamination" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Ambient contamination<a href="preprocess.html#ambient-contamination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I tested the combinations of the aforementioned empty droplets and ambient RNA detection methods (See diagram) with our in-house data and I’m convinced that Cellbender and cluster-based methods are of advantage.</p>
<p>In snRNA-seq, mtRNAs are natural controllers for ambient RNAs and thus can be leveraged in benchmarking.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rna-dynamics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/brainfo/brainfo.github.io/edit/master/01-Preprocess.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
